lib/fast_gettext/vendor/poparser.rb

```diff
@@ -132,14 +132,9 @@ def on_comment(comment)
   end
 
   def parse_file(po_file, data)
-    args = [ po_file ]
-    # In Ruby 1.9, we must detect proper encoding of a PO file.
-    if String.instance_methods.include?(:encode)
-      encoding = detect_file_encoding(po_file)
-      args << "r:#{encoding}"
-    end
     @po_file = po_file
-    parse(File.open(*args) {|io| io.read }, data)
+    encoding = detect_file_encoding(po_file)
+    parse(File.open(po_file, "r:#{encoding}") {|io| io.read }, data)
   end
 
   def detect_file_encoding(po_file)
```

Comment:
I'd love to.

The "proper" ruby way seems to need reworking the whole parser to work in binmode, extract the charset properly, and then tag all strings (force_encoding) before returning them to the world. I don't know what that would break downstream, nor do I currently have the cycles this week to help out with the work :(